### PR TITLE
migrate from kapt to ksp

### PR DESCRIPTION
**Pull request by patrickunterwegs**
- [x] TODO: check if action is really necessary here: 

> The plugin will then configure the Room compiler and the various compile tasks and its backends (javac, KAPT, KSP) to output schema files into flavored folders, i.e. schemas/flavorOneDebug/com.package.MyDatabase/1.json. As usual these files are checks-in into the repository to be used for validation and auto-migrations. Upon migrating to using the plugin instead of the annotation processor options the existing schema files must be copied to the generated flavor directories created by the plugin, this is a one-time migration operation that must be done manually. The schema documentation in [developers.android.com](https://developer.android.com/) will be updated in the future once feedback is addressed and the plugin reaches stable, so please give it a try.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -57,6 +57,10 @@ buildscript {
     }
 }
 
+plugins {
+    id 'com.google.devtools.ksp' version '1.9.0-1.0.12' apply false
```
Comment (andrekir) [inline build.gradle]: '1.9.0-1.0.12' is for Kotlin 1.9.0, based on your `version_kotlin` you need '1.8.22-1.0.11' ("$version_kotlin-1.0.11" also works).

File: app/build.gradle [hunk 2]
```diff
@@ -39,10 +41,12 @@ android {
...
-        // defining the room.schemaLocation explicitly allows AutoMigrations
-        kapt {
-            arguments {
-                arg("room.schemaLocation", "$projectDir/schemas")
+
+        ksp {
+            // defining the room.schemaLocation explicitly allows AutoMigrations
+            room {
+                generateKotlin
+                schemaDirectory "$projectDir/schemas/"
```
Comment (andrekir) [inline app/build.gradle]: try:

```
ksp {
    arg("room.generateKotlin", "true")
    arg("room.schemaLocation", "$projectDir/schemas")
}
```

File: build.gradle (hunk 1 above)
Comment (patrickunterwegs) [reply build.gradle]: Thank you very much for checking it out and your hint, I downgraded and I could continue =)

File: app/build.gradle (hunk 2 above)
Comment (patrickunterwegs) [reply app/build.gradle]: Thanks a lot, it worked =)

Do you maybe also know where this needs to be resolved: 
`e: [ksp] Entities and POJOs must have a usable public constructor. You can have an empty constructor or a constructor whose parameters match the fields (by name and type). - android.database.Cursor`

Regrettably there are no further details, I tried around for a while and searched on the internet, but I couldn't find anything. Or do you maybe know any working reference implementation?

Comment (andrekir) [reply app/build.gradle]: I got nothing on this one unfortunately. `android.database.Cursor` is used in [ICalDatabaseDao](
https://github.com/TechbeeAT/jtxBoard/blob/develop/app/src/main/java/at/techbee/jtx/database/ICalDatabaseDao.kt#L735) and [SyncContentProvider](https://github.com/TechbeeAT/jtxBoard/blob/develop/app/src/main/java/at/techbee/jtx/SyncContentProvider.kt#L360).

maybe worth asking in the ksp repo if there are any known issues with `android.database.Cursor` or Simple/SupportSQLiteQuery: https://github.com/google/ksp/issues

and thank you for the awesome app. =)

Comment (patrickunterwegs) [reply app/build.gradle]: Thanks a lot =)
In fact, with your comment I reviewed it again and came to the right place, it works now =)

**Comment by patrickunterwegs** [comment]
Migration check successful (migrated from 2.1.x to the current one and no DB issues occurred

**Comment by patrickunterwegs** [comment]
all checks successful, ready to merge

**Pull request merged**